src/v/raft/service.h

```diff
@@ -137,16 +138,23 @@ class service final : public raftgen_service {
           = co_await ss::when_all_succeed(futures.begin(), futures.end());
 
         heartbeat_reply_v2 reply(_self, source);
-
+        ssx::async_counter cnt;
         // flatten responses
         for (shard_heartbeat_replies& shard_replies : replies) {
-            for (const auto& lw_reply : shard_replies.lw_replies) {
-                reply.add(lw_reply.group, lw_reply.result);
-            }
-            for (const auto& full_reply : shard_replies.full_heartbeats) {
-                reply.add(full_reply.group, full_reply.result, full_reply.data);
-            }
-            co_await ss::coroutine::maybe_yield();
+            co_await ssx::async_for_each_counter(
+              cnt,
+              shard_replies.lw_replies.begin(),
+              shard_replies.lw_replies.end(),
+              [&reply](lw_reply& lw_reply) {
+                  reply.add(lw_reply.group, lw_reply.result);
+              });
+            co_await ssx::async_for_each_counter(
+              cnt,
+              shard_replies.full_heartbeats.begin(),
+              shard_replies.full_heartbeats.end(),
+              [&reply](full_heartbeat_reply& full_reply) {
+                  reply.add(full_reply.group, full_reply.result);
```

Comment:
why does the reply not contain `full_reply.data`?

Reply:
great catch